lsst_dashboard/dataset.py

```diff
@@ -118,3 +132,125 @@ def fetch_visits_by_metric(self):
+            self.failures = self.metadata.get('failures', {})
+            if self.tracts is None:
+                self.tracts = list(set(x for v in self.metadata['visits'].values() for x in v.keys()))
+
+    def fetch_coadd_table(self, coadd_version='unforced'):
+        table = 'qaDashboardCoaddTable'
+        store = partial(get_store_from_url, 'hfs://' + str(self.path))
+        print(str(self.path))
+        predicates = [[('tract', 'in', self.tracts)]]
+        dataset = "coadd_{}".format(coadd_version)
+
+        coadd_df = read_dataset_as_ddf(predicates=predicates,
+                                       dataset_uuid=dataset,
+                                       store=store,
+                                       table='table').compute()
+
+        # hack label in ...
+        coadd_df['label'] = 'star'
+
+        self.coadd[table] = coadd_df
+
+    def post_process_metadata(self):
+        df = self.coadd['qaDashboardCoaddTable']
+        self.flags = df.columns[df.dtypes == bool].to_list()
+        self.filters = list(self.metadata['visits'].keys())
+        self.metrics = (set(df.columns.to_list()) -
+                        set(self.flags) -
+                        set(['patch', 'dec', 'label',
+                             'psfMag', 'ra', 'filter',
+                             'dataset', 'dir0', 'tract']))
+
+    def fetch_visits(self):
+        store = partial(get_store_from_url, 'hfs://' + str(self.path))
+        predicates = [[('tract', 'in', self.tracts)]]
+        self.visits = read_dataset_as_ddf(predicates=predicates,
+                                          dataset_uuid='visits',
+                                          store=store,
+                                          table='table')
+
+    def get_visits_by_metric_filter(self, filt, metric):
+
+        store = partial(get_store_from_url, 'hfs://' + str(self.path))
+
+        columns = ['filter', 'tract', 'visit', 'calib_psf_used',
+                   'calib_psf_candidate', 'calib_photometry_reserved',
+                   'qaBad_flag', 'ra', 'dec', 'psfMag'] + [metric]
+
+        visits_ddf = read_dataset_as_ddf(dataset_uuid="visits",
+                                         predicates=[[('filter', '==', filt)]],
+                                         store=store,
+                                         columns=columns,
+                                         table='table')
+        # hack label in ...
+        visits_ddf['label'] = 'star'
+
+        return visits_ddf[visits_ddf[metric].notnull()]
+
+    def fetch_visits_by_metric(self):
+        for filt in self.filters:
+            self.visits_by_metric[filt] = {}
+            for metric in self.metrics:
+                try:
+                    ddf = self.get_visits_by_metric_filter(filt, metric)
+                except:
+                    print('WARNING: problem loading visits for {} metric and {} filter'.format(metric, filt))
+                    ddf = None
```

Comment:
Maybe print a traceback here, too?  Or tighten the `except` to catch precisely what happens when the metric doesn't exist in the visit data?

Reply:
@timothydmorton yes, I think at least using `traceback.print_exc()` would help the cause.  The PR is getting large so I'm going to hit these items in a follow up